<a href="https://colab.research.google.com/github/sehernandezce/AlgorithmsUN2022II/blob/main/Quiz6/sehernandezce_Quiz6.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Group 9 Quiz 6

## Members

1.   De La Torre Sanchez, Juan Manuel
2.   Hernandez Ceron, Sebastian
3.   Yepes Garcia, Brayan Sebastian



**The short-term values are**: 6, 12, 24.

**The long-term values are**: 101, 186, 240.

**At 6 months:**

data = qndata.stocks.load_ndx_data(min_date="2021-09-01")

stats = qns.calc_stat(data, weights.sel(time=slice("2022-04-14", None)))

**At 12 months:**

data = qndata.stocks.load_ndx_data(min_date="2019-09-01")

stats = qns.calc_stat(data, weights.sel(time=slice("2020-10-14", None)))

In [1]:
import pandas as pd  
  
# assign data of lists.  
data = {'LMA': [240, 240, 240,186,186,186,101,101,101], 
        'SMA': [24,12,6,24,12,6,24,12,6] , 
        'R6M': [0.2964611585853113, 0.2773011020333751, 0.27597973719398033, 0.19446299500032616, 0.2122831033640267, 0.18752027644469543,-0.6643084148809723, -0.6075206292049692, -0.57168119602436], 
        'R2Y': [0.3344227718817993, 0.2480232855875463, 0.25139733519608204, 0.26782458694155853, 0.14603399754674593, 0.13617480109046007, -0.0658641091967502, -0.14756460131656665,-0.2439454353235792]}  
  

# Create DataFrame  
df = pd.DataFrame(data)  

df['AR'] = (df['R6M'] + df['R2Y'])/2

df = df.sort_values(by=['AR'], ascending=False)
  
df

LMA  SMA       R6M       R2Y        AR
0  240   24  0.296461  0.334423  0.315442
2  240    6  0.275980  0.251397  0.263689
1  240   12  0.277301  0.248023  0.262662
3  186   24  0.194463  0.267825  0.231144
4  186   12  0.212283  0.146034  0.179159
5  186    6  0.187520  0.136175  0.161848
6  101   24 -0.664308 -0.065864 -0.365086
7  101   12 -0.607521 -0.147565 -0.377543
8  101    6 -0.571681 -0.243945 -0.407813

***Result***

In quiz 6 it is observed that the best values were:

Slow = 240 Fast = 24 AR = 0.315442

In this case, during this time it is profitable to buy.
